x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -47,19 +68,29 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       : TimeoutErrorMode.CONTACT;
   }
 
-  public search({ id, ...request }: IKibanaSearchRequest, options: IAsyncSearchOptions = {}) {
-    const searchOptions = {
-      strategy: ENHANCED_ES_SEARCH_STRATEGY,
-      ...options,
+  private createRequestHash$(request: IKibanaSearchRequest, options: IAsyncSearchOptions) {
+    const { sessionId, isRestore } = options;
+    const { preference, ...params } = request.params || {};
```

Comment:
What is `preference` and why do we handle it differently? 
Looks like `request.params` are `any` and we rely on some "knowledge" outside of this module?

Reply:
preference is a random number so we exclude it as it always breaks the hash.
And yes, we have some `any`s here :-| but I'd like to keep it out of this PR

Reply:
Ok, got it! Comment might be helpful then

Reply:
pls add a comment with this link